docs/common-patterns.rst

```diff
@@ -133,6 +141,21 @@ restrictions highly readable.
         address public owner = msg.sender;
         uint public creationTime = block.timestamp;
 
+        // Now follows a list of errors that
+        // this contract can generate together
+        // with a textual explanation in special
+        // comments.
+
+        /// Sender not authorized for this
+        /// operation.
+        error Unauthorized();
```

Comment:
Somehow I think it is better to also encode the "namespace" into the error. Why would it not make sense to encode where exactly the error was defined? E.g. the json for this could look like 
```json
[{
  "type":"error",
  "inputs": [],
  "name":"AccessRestriction::Unauthorized"
  },
...
]
```
If an error was defined as a "free-error" (not sure, is it supported?), there will be no namespace added.
This would also change the signature e.g. instead of `Unauthorized()`, `AccessRestriction::Unauthorized()` could be used. I think this will reduced potential clashes if e.g. Libraries are used. 

I think it would be important to not just define all errors (and also events) in the "global namespace". I think it would be important to distinguish different errors (and events) that may have the same name, but where defined in different places.

@hrkrshnn @chriseth

Reply:
After looking at #9011 I think it would also make sense to add such "namespaces" for the events. But maybe these "namespaces" make only sense, if events or errors where defined in libraries.